### PR TITLE
Add container metrics and grafana dashboard

### LINKED ISSUE #2633: Reintroduce Kubernetes cluster dashboard
## Background

When debugging application issues it's important to have observability into the Kubernetes cluster as a whole, specifically memory and cpu usage per-node and per-container. This information, alongside Linkerd's request-aware metrics, is helpful to eliminate (or identify) resource contention as a root cause.

This functionality was added via #904 and #1142. It was subsequently removed in #1410, as the focus shifted away from cluster owners to service owners. Also the RBAC permissions for node-level metrics collection were incongruent with permissions available to service owners.

## Proposal

Reintroduce a Kubernetes cluster dashboard. Use #904 and #1142 as a starting point, but survey current state of the art Kubernetes Grafana dashboards at https://grafana.com/dashboards.

## Open questions

This change will require additional RBAC privileges, most likely the `get` verb on `nodes` and `nodes/proxy` resources. We should evaluate this tradeoff (and if other options for cluster monitoring require fewer/different permissions). If we do move forward, it will have implications on `linkerd check`'s permissions checks.

### PR DESCRIPTION
**Pull request by grampelberg**
Fixes #2633.

### CODE REVIEW

File: chart/templates/prometheus.yaml [hunk 1]
```diff
@@ -40,6 +40,25 @@ data:
...
 
+    - job_name: 'kubernetes-nodes-cadvisor'
+      scheme: https
+      tls_config:
+        ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
+        insecure_skip_verify: true
+      bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
+
+      kubernetes_sd_configs:
+        - role: node
```
[the remark this answers is not in the document]
Comment (siggy) [reply chart/templates/prometheus.yaml]: oh! also reference https://grafana.com/grafana/dashboards/315 immediately above `job_name: 'kubernetes-nodes-cadvisor'`

File: charts/linkerd2/templates/prometheus-rbac.yaml [hunk 2]
```diff
@@ -13,7 +13,7 @@ metadata:
     {{.ControllerNamespaceLabel}}: {{.Namespace}}
 rules:
 - apiGroups: [""]
-  resources: ["pods"]
+  resources: ["nodes", "nodes/proxy", "pods"]
```
[the remark this answers is not in the document]
Comment (grampelberg) [reply charts/linkerd2/templates/prometheus-rbac.yaml]: I don't *think* it does? We're not changing what the user installing the control plane requires, just what permissions the control plane has.

File: grafana/dashboards/kubernetes.json [hunk 3]
```diff
@@ -0,0 +1,2079 @@
+{
+  "__inputs": [
```
[the remark this answers is not in the document]
Comment (grampelberg) [reply grafana/dashboards/kubernetes.json]: I'd like to have these in as it makes putting the dashboards up on grafana hub easier and doesn't appear to have any impact. It probably makes sense to add those in as part of that PR though.

[the remark this answers is not in the document]
Comment (siggy) [reply grafana/dashboards/kubernetes.json]: Makes sense, yeah, in that case recommend adding these bits to the top of every dashboard.

File: charts/linkerd2/templates/prometheus-rbac.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (siggy) [reply charts/linkerd2/templates/prometheus-rbac.yaml]: For reference, this is the error I've bumped into when trying to create things with broader RBAC privileges than the thing I'm trying to create it with:
https://github.com/kubernetes/kubernetes/blob/282b992e0c1ffa08a4ede0cca7cec06c95da5ce3/pkg/registry/rbac/validation/rule.go#L80

[the remark this answers is not in the document]
Comment (grampelberg) [reply charts/linkerd2/templates/prometheus-rbac.yaml]: After doing some research, it appears that users either need `*` or `escalate` in the verb section. As we're not testing comprehensively right now, I feel okay merging this but I've created #3224 to track discussion on a way to be more confident that things will work correctly.

**Pull request merged**